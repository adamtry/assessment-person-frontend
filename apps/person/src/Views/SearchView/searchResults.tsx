import React, { useState, useEffect } from "react";
import { personSearchResult } from "../../Interfaces";
import { Pagination } from "../../Components";
import { SearchResultsGroup } from "../../Components/SearchResultsGroup";

interface myProps {
  results: personSearchResult[];
  maxSearchResults: number;
}

export const SearchResults = (props: myProps): JSX.Element => {
  const sliceIntoChunks = (arr: any, chunkSize: number) => {
    const res = [];
    for (let i = 0; i < arr.length; i += chunkSize) {
      const chunk = arr.slice(i, i + chunkSize);
      res.push(chunk);
    }
    return res;
  };
  const [results, setResults] = useState<personSearchResult[]>(
    sliceIntoChunks(props.results, props.maxSearchResults)[0]
  );

  useEffect(() => {
    setResults(sliceIntoChunks(props.results, props.maxSearchResults)[0]);
  }, [props.results, props.maxSearchResults]);

  const [splitResults] = useState<personSearchResult[][]>(
    sliceIntoChunks(props.results, props.maxSearchResults)
  );

  const onPageChange = (currentPage: number, isNext: boolean) => {
    if (isNext) {
      setResults(splitResults[currentPage]);
    } else {
      setResults(splitResults[currentPage - 2]);
    }
  };

  let numberOfResults = props.results.length;

  return (
    <div className="govuk-grid-row">
      <div className="govuk-grid-column-two-thirds">
        <div className="sv-group">
          <h2 className="lbh-heading-h3 govuk-!-margin-top-7">{`${numberOfResults} results found`}</h2>
        </div>
        <hr />
        <div id="searchResults">
          {results &&
            results.length > 0 && [<SearchResultsGroup results={results} />]}
        </div>
        {numberOfResults > 0 && (
          <Pagination
            total={props.results.length}
            onPageChange={onPageChange}
            pageSize={props.maxSearchResults}
          />
        )}
      </div>
    </div>
  );
};
