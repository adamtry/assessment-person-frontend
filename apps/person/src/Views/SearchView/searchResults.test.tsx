import React from 'react'
import { render } from "@testing-library/react";
import { SearchResults } from "./searchResults";

import { MemoryRouter } from 'react-router-dom';

beforeEach(() => {
    jest.resetModules();
});

describe("Search Results", () => {

    test("should render search results", () => {
        const wrapper = render(
            <MemoryRouter>
                <SearchResults key={0} maxSearchResults={10} results={[]} />
            </MemoryRouter>
        );
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.getByText("0 results found")).toBeInTheDocument();
    });
});