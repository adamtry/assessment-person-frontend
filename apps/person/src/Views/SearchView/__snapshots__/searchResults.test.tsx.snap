// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Results should render search results 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="govuk-grid-row"
      >
        <div
          class="govuk-grid-column-two-thirds"
        >
          <div
            class="sv-group"
          >
            <h2
              class="lbh-heading-h3 govuk-!-margin-top-7"
            >
              0 results found
            </h2>
          </div>
          <hr />
          <div
            id="searchResults"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="govuk-grid-row"
    >
      <div
        class="govuk-grid-column-two-thirds"
      >
        <div
          class="sv-group"
        >
          <h2
            class="lbh-heading-h3 govuk-!-margin-top-7"
          >
            0 results found
          </h2>
        </div>
        <hr />
        <div
          id="searchResults"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
